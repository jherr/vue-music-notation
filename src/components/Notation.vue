<template>
  <div
    class="container"
  >
    <Staff
      :showTrebleClef="showTrebleClef"
      :showBassClef="showBassClef"
      :showBrace="true"
      :showEnd="true"
    />
    <slot />
    <Staff
      :showEnd="true"
    />
  </div>
</template>

<script>
import Staff from './Staff';
export default {
  props: {
    showTrebleClef: {
      default: true,
    },
    showBassClef: {
      default: true,
    },
  },
  components: {
    Staff,
  },
  provide() {
    return {
      showTrebleClef: this.showTrebleClef,
      showBassClef: this.showBassClef,
    };
  }
}
</script>

<style scoped>
.container {
  font-size: 0;
}
</style>
