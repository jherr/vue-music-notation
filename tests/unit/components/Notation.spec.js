import { mount } from '@vue/test-utils';
import Notation from '@/components/Notation.vue';

describe('Notation', () => {
  it('renders', () => {
    const wrapper = mount(Notation, {
      propsData: {
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
