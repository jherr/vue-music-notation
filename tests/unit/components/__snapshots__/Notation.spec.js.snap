// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notation renders 1`] = `
<div
  class="container"
>
  <svg
    class="container"
    height="180"
    showBassClef="true"
    showTrebleClef="true"
    width="45"
  >
    <line
      style="stroke: black; stroke-width: 3;"
      x1="1"
      x2="1"
      y1="30"
      y2="70"
    />
    <line
      style="stroke: black; stroke-width: 3;"
      x1="1"
      x2="1"
      y1="110"
      y2="150"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="70"
      y2="70"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="60"
      y2="60"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="50"
      y2="50"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="40"
      y2="40"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="30"
      y2="30"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="110"
      y2="110"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="120"
      y2="120"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="130"
      y2="130"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="140"
      y2="140"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="45"
      y1="150"
      y2="150"
    />
     
    <g
      transform="translate(5,34) scale(0.028,0.028)"
    >
      <path
        d="M 592.10873,1275.9669 C 461.75172,1268.3902 328.65904,1186.6265 249.0601,1092.783 C 156.77394,983.97782 118.72592,836.04683 128.47199,714.56357 C 157.10277,357.61288 545.27831,146.63848 688.97108,-9.280262 C 785.15294,-113.64625 805.31643,-164.52308 826.79977,-218.19949 C 868.39181,-322.09965 875.09166,-443.8341 792.63375,-452.92251 C 713.90712,-461.59988 649.13737,-337.79201 620.20973,-253.17845 C 594.19587,-177.07331 576.90507,-100.71696 592.5563,13.979673 C 599.58954,65.50958 793.18636,1503.9125 796.45179,1526.2088 C 829.05589,1749.0255 701.63092,1841.2249 571.55248,1857.6251 C 290.65671,1893.038 200.52617,1607.5843 326.4212,1499.1719 C 423.34291,1415.7001 564.35026,1487.3615 556.73245,1624.5919 C 549.98693,1746.1391 430.80546,1749.7197 400.35244,1746.9429 C 447.10065,1830.7846 799.52998,1874.5871 745.41513,1495.7923 C 737.811,1442.5634 558.91549,90.842953 554.53112,60.595454 C 521.71238,-165.84753 516.71147,-345.08557 634.69182,-554.25141 C 678.24767,-631.46637 747.0821,-681.3156 780.87362,-674.7893 C 788.29962,-673.35526 795.69824,-670.62872 801.57144,-664.56827 C 892.07191,-571.31845 919.83494,-364.53202 909.9199,-245.74332 C 899.76736,-124.11391 894.1088,1.7993735 773.16902,148.63428 C 726.36601,205.45738 583.54553,330.63538 501.65851,402.55255 C 386.60107,503.59831 303.14756,591.85179 257.99323,698.31862 C 207.24886,817.97506 198.65826,968.6006 313.27268,1102.2505 C 379.20247,1177.7619 488.59222,1231.3424 580.65459,1232.4842 C 836.63719,1235.6628 911.39048,1109.4801 913.77904,966.58197 C 917.71126,731.28351 633.64596,642.32214 516.85762,804.10953 C 449.14212,897.92109 478.90552,996.66049 524.38411,1043.6371 C 539.99424,1059.7587 557.43121,1072.0395 573.92734,1078.8855 C 579.9056,1081.3654 593.96751,1087.9054 589.97593,1097.4779 C 586.6557,1105.4428 580.20702,1105.8904 574.33381,1105.1871 C 500.68573,1096.3544 419.13667,1025.958 399.0828,904.87212 C 369.86288,728.38801 525.6035,519.0349 747.9133,553.274 C 893.45572,575.68903 1028.5853,700.92182 1016.7338,934.11946 C 1006.5722,1133.9822 840.87996,1290.4262 592.10873,1275.9669 z"
      />
    </g>
     
    <g
      transform="translate(10,110) scale(1.7,1.7)"
      viewBox="0 0 18 20"
    >
      <g
        transform="translate(6.600000e-3,3.125356e-3)"
      >
        <g
          style="fill: #000000; fill-opacity: 1.0000000;"
          transform="translate(-230.9546,-533.6597)"
        >
          <path
            d="M 248.25999,536.80200 C 248.26766,537.17138 248.11044,537.54065 247.82878,537.78185 C 247.46853,538.11076 246.91933,538.17813 246.47048,538.01071 C 246.02563,537.83894 245.69678,537.39883 245.67145,536.92060 C 245.63767,536.54689 245.75685,536.15479 246.02747,535.88867 C 246.28257,535.61680 246.66244,535.48397 247.03147,535.50645 C 247.41131,535.51452 247.77805,535.70601 248.00489,536.01019 C 248.17962,536.23452 248.26238,536.51954 248.25999,536.80200 z "
            style="fill: #000000; fill-opacity: 1.0000000;"
          />
        </g>
         
        <g
          style="fill: #000000; fill-opacity: 1.0000000;"
          transform="translate(-230.9546,-533.6597)"
        >
          <path
            d="M 248.25999,542.64502 C 248.26772,543.01469 248.11076,543.38446 247.82878,543.62585 C 247.46853,543.95476 246.91933,544.02213 246.47048,543.85472 C 246.02537,543.68288 245.69655,543.24237 245.67145,542.76389 C 245.63651,542.38990 245.76354,542.00308 246.02700,541.73300 C 246.27663,541.45454 246.66060,541.32790 247.02845,541.34950 C 247.51230,541.36282 247.95159,541.69251 248.15162,542.12465 C 248.22565,542.28740 248.26043,542.46657 248.25999,542.64502 z "
            style="fill: #000000; fill-opacity: 1.0000000;"
          />
        </g>
         
        <g
          style="fill: #000000; fill-opacity: 1.0000000;"
          transform="translate(-230.9546,-533.6597)"
        >
          <path
            d="M 243.97900,540.86798 C 244.02398,543.69258 242.76360,546.43815 240.76469,548.40449 C 238.27527,550.89277 235.01791,552.47534 231.69762,553.53261 C 231.25590,553.77182 230.58970,553.45643 231.28550,553.13144 C 232.62346,552.52289 234.01319,552.00050 235.24564,551.18080 C 237.96799,549.49750 240.26523,546.84674 240.82279,543.61854 C 241.14771,541.65352 241.05724,539.60795 240.56484,537.67852 C 240.20352,536.25993 239.22033,534.79550 237.66352,534.58587 C 236.25068,534.36961 234.74885,534.85905 233.74057,535.88093 C 233.47541,536.14967 232.95916,536.89403 233.04435,537.74747 C 233.64637,537.27468 233.60528,537.32732 234.09900,537.10717 C 235.23573,536.60031 236.74349,537.32105 237.02700,538.57272 C 237.32909,539.72295 237.09551,541.18638 235.96036,541.79960 C 234.77512,542.44413 233.02612,542.17738 232.36450,540.90866 C 231.26916,538.95418 231.87147,536.28193 233.64202,534.92571 C 235.44514,533.42924 238.07609,533.37089 240.19963,534.13862 C 242.38419,534.95111 243.68629,537.21483 243.89691,539.45694 C 243.95419,539.92492 243.97896,540.39668 243.97900,540.86798 z "
            style="fill: #000000; fill-opacity: 1.0000000;"
          />
        </g>
      </g>
    </g>
      
  </svg>
    
  <svg
    class="container"
    height="180"
    width="3"
  >
    <line
      style="stroke: black; stroke-width: 3;"
      x1="1"
      x2="1"
      y1="30"
      y2="70"
    />
    <line
      style="stroke: black; stroke-width: 3;"
      x1="1"
      x2="1"
      y1="110"
      y2="150"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="70"
      y2="70"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="60"
      y2="60"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="50"
      y2="50"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="40"
      y2="40"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="30"
      y2="30"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="110"
      y2="110"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="120"
      y2="120"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="130"
      y2="130"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="140"
      y2="140"
    />
    <line
      style="stroke: black; stroke-width: 1;"
      x1="0"
      x2="3"
      y1="150"
      y2="150"
    />
     
    <!---->
     
    <!---->
      
  </svg>
</div>
`;
